src/Core/Settings.h

```diff
@@ -346,6 +346,7 @@ class IColumn;
     \
     M(UInt64, max_rows_in_join, 0, "Maximum size of the hash table for JOIN (in number of rows).", 0) \
     M(UInt64, max_bytes_in_join, 0, "Maximum size of the hash table for JOIN (in number of bytes in memory).", 0) \
+    M(UInt64, max_merged_block_size, 0, "Maximum size of the merged block for JOIN (in number of bytes in memory).", 0) \
```

Comment:
Isn't it the same as `max_joined_block_size_rows` (but it bytes not rows)?

Reply:
I think we can limit the bytes to avoiding memory overflow， but rows can not do that.